routers/web/repo/issue.go

```diff
@@ -2471,7 +2471,7 @@ func addParticipant(poster *user_model.User, participants []*user_model.User) []
 
 func filterXRefComments(ctx *context.Context, issue *models.Issue) error {
 	// Remove comments that the user has no permissions to see
-	for i := 0; i < len(issue.Comments); {
+	for i := 0; i < len(issue.Comments); i++ {
```

Comment:
I think this is wrong.

Look at line 2488.

Reply:
should be no difference https://go.dev/play/p/Ge-Q9rs_930

or what do you refer to?

Reply:
> should be no difference https://go.dev/play/p/Ge-Q9rs_930
> 
> or what do you refer to?

Well, for go it does. As `i++` isn't always run in the second case(in this one it does), but it's harder to prove that, so it currently fails to do that. With this the first patch, it does know that the `i++` will run and that the code doesn't do weird things with the `i`. So it can say that anything e.g. if you're iterating over a slice, won't require any bound checks, so increases performance less binary size etc. But that isn't the reason for this patch, go doesn't like it that the `i++` is the last statement and that their are `continue` within the scope, so it's currently throwing a lot of "info" here(it's a bug on go's side btw), so for now this should fix that and I can happily traverse the output again 😄.